app/src/main/java/com/example/cinemapp/data/MovieRemoteDataSource.kt

```diff
@@ -95,16 +96,40 @@ interface MovieRemoteDataSource {
         @Body body: SessionRequestDTO
     ): Response<SessionResponseDTO>
 
-    @DELETE("$API_VERSION/authentication/session")
+    @HTTP(method = "DELETE", path = "$API_VERSION/authentication/session", hasBody = true)
     suspend fun deleteSession(
-        @Body raw: JsonObject
+        @Body body: SessionDeleteBodyDTO
     ): Response<SessionDeleteResponseDTO>
 
     @GET("$API_VERSION/account")
     suspend fun getAccountDetails(
         @Query("session_id") sessionId: String
     ): Response<AccountDetailsDTO>
 
+    @GET("$API_VERSION/account/account_id/favorite/movies")
+    suspend fun getFavorite(
+        @Query("language") language: String = "en-US",
+        @Query("page") page: Int = 1,
+        @Query("session_id") sessionId: String,
+        @Query("sort_by") sortBy: String = "created_at.desc"
+    ): Response<MovieResponseDTO>
+
+    @GET("$API_VERSION/account/account_id/watchlist/movies")
+    suspend fun getWatchlist(
+        @Query("language") language: String = "en-US",
+        @Query("page") page: Int = 1,
+        @Query("session_id") sessionId: String,
+        @Query("sort_by") sortBy: String = "created_at.desc"
+    ): Response<MovieResponseDTO>
+
+    @GET("$API_VERSION/account/account_id/rated/movies")
+    suspend fun getRated(
+        @Query("language") language: String = "en-US",
```

Comment:
You can create `companion object` here, just so some you do not have [Magic string](https://en.wikipedia.org/wiki/Magic_string#:~:text=In%20computer%20programming%2C%20a%20magic,expected%20response%20in%20most%20situations.). You can call it something like `const val DEFAULT_LANGUAGE`.

Same stands for `sortBy` parameter